examples/src/main/java/tachyon/examples/JournalCrashTest.java

```diff
@@ -0,0 +1,387 @@
+    sClientThreadList = new ArrayList<ClientThread>();
+
+    // Start Tachyon Master and prepare for this test.
+    System.out.println("Start Journal Crash Test...");
+    startMaster();
+    sTfs = TachyonFileSystem.get();
+    try {
+      sTfs.delete(sTfs.open(new TachyonURI(sTestDir)));
+    } catch (Exception ioe) {
+      // Test Directory not exist
+    }
+
+    // Setup and launch all the client threads.
+    for (int i = 0; i < sCreateFileClientNum; i ++) {
+      ClientThread thread = new ClientThread(sTestDir + "/createFile" + i + "/",
+          ClientOpType.CREATE_FILE);
+      sClientThreadList.add(thread);
+    }
+    for (int i = 0; i < sCreateDeleteClientNum; i ++) {
+      ClientThread thread = new ClientThread(sTestDir + "/createDelete" + i + "/",
+          ClientOpType.CREATE_DELETE_FILE);
+      sClientThreadList.add(thread);
+    }
+    for (int i = 0; i < sCreateRenameClientNum; i ++) {
+      ClientThread thread = new ClientThread(sTestDir + "/createRename" + i + "/",
+          ClientOpType.CREATE_RENAME_FILE);
+      sClientThreadList.add(thread);
+    }
+    for (Thread thread : sClientThreadList) {
+      thread.start();
+    }
+
+    // Kill/Restart master for several times.
+    System.out.println("Kill/Restart master for " + sCrashTimes + " times...");
+    CommonUtils.sleepMs(LOG, 1000);
+    for (int i = 0; i < sCrashTimes; i ++) {
+      System.out.println("\tTimes " + i);
+      killMaster();
+      startMaster();
+    }
+    killMaster();
+
+    // Wait all client threads stopped.
+    System.out.println("Wait all the Clients stopped...");
+    System.out.println("(This step may take a few seconds due to the retry policy)");
+    for (ClientThread clientThread : sClientThreadList) {
+      clientThread.setIsStopped(true);
+    }
+    for (Thread thread : sClientThreadList) {
+      try {
+        thread.join();
+      } catch (InterruptedException e) {
+        LOG.error("Error when waiting thread", e);
+      }
+    }
+
+    // Restart Tachyon Master. Check status and print pass info.
+    System.out.println("Check status...");
+    startMaster();
+    boolean checkSuccess = false;
+    try {
+      checkSuccess = checkStatus();
+    } catch (Exception e) {
+      LOG.error("Failed to check status", e);
+    } finally {
+      stopCluster();
+    }
+    Utils.printPassInfo(checkSuccess);
+    System.exit(checkSuccess ? EXIT_SUCCESS : EXIT_FAILED);
+  }
+
+  /**
+   * Parse the input args with a command line format, using
+   * <code>org.apache.commons.cli.CommandLineParser</code>.
+   * @param args the input args
+   * @return true if parse successfully and can run the next step, false if parse failed or need to
+   * print out the help information.
+   */
+  private static boolean parseInputArgs(String[] args) {
+    Options options = new Options();
+    options.addOption("help", false, "Show help for this test");
+    options.addOption("crashes", true, "Repeat times to kill/restart Master");
```

Comment:
I think the way crashes are done now will not stress journal in a variety of situations, since the crashes all happen right after another. I think another way to do this is to define the total time to run the test for, and the maximum time a master should ever be alive. This means I could run the test for 24 hours, and if I want the master to be up for at most 1 hour, the test will kill the master after a random time (less than an hour). This means in the 24 hour test, I know the master would have been killed at least 24 times, but with random intervals.